modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/execution.scala

```diff
@@ -0,0 +1,71 @@
+package edu.gemini.seqexec.engine
+
+import scalaz._
+import Scalaz._
+
+/**
+  * This structure holds the current `Execution` under execution. It carries
+  * information about which `Action`s have been completed.
+  *
+  */
+case class Current(execution: Execution[Action \/ Result]) {
```

Comment:
We discussed this a bit on the chat. The type `Action \/ Result` seems to me like `Error \/ Ok` while in fact errors will occur in `Result`. Probably doesn't matter but maybe worth some discussion